torch/distributed/_composable/fsdp/_fsdp_param.py

```diff
@@ -162,6 +162,7 @@ def __init__(
         self._init_extensions()
         self.all_gather_outputs: List[torch.Tensor] = []
         self.unsharded_accumulated_grad = None
+        self._unsharded_param = None
```

Comment:
probably need to change this type annotation, but it may require adding more `assert ... is not None` to make mypy happy 😢 :
https://github.com/pytorch/pytorch/blob/996ce17de57cce3b4ae50466d88703490ff810f7/torch/distributed/_composable/fsdp/_fsdp_param.py#L128

Reply:
yes you are right - those asserts are so annoying to add and I gave up... not having it as `Optional[]` is so much cleaner